docs/developers/api.md

```diff
@@ -4,9 +4,24 @@ title: API Overview
 sidebar_label: Overview
 ---
 
+## Compatibility considerations
+
+[`scalafix-core`](https://index.scala-lang.org/scalacenter/scalafix/scalafix-core) makes the packages
+below available to built-in and external rules.
+
+Its A.B.C scheme follows [PVP](https://pvp.haskell.org/) for syntactic and semantic binary compatiblity
```

Comment:
I find the wording a bit confusing. It seems that you talk about “semantic binary compatibility” and “syntactic binary compatibility”. Although I guess you really talk about the binary compatibility of the API for syntactic and semantic rules?

```suggestion
Its A.B.C version scheme follows [PVP](https://pvp.haskell.org/) for binary compatiblity
```

Reply:
I agree this was confusing, particularly in the context of scalafix where we have syntactic and semantic rules - I'll amend with your wording.

For the record, what I meant was that a major version will be bumped even though no linkage error may happen, but just because the behavior/semantics changed. Binary compatibility, as described in https://docs.scala-lang.org/overviews/core/binary-compatibility-for-library-authors.html for example, seems to focus on link guarantees, so I wanted to highlight the silent functional changes as a factor.

Reply:
> a major version will be bumped even though no linkage error may happen, but just because the behavior/semantics changed

In such a case, would it be possible to not change the semantics of the existing methods but to add new methods instead?

Reply:
That's what we aim for, but we can control dependencies that are [part of the API](https://scalacenter.github.io/scalafix/docs/developers/api.html#scalameta-trees): https://github.com/scalameta/scalameta/pull/2601 is what resulted in a (silent) breaking change, identified by the [community build](https://github.com/scalacenter/scalafix/pull/1556).